src/wallet/api/stake_unlock_result.cpp

```diff
@@ -0,0 +1,36 @@
+#include "stake_unlock_result.h"
+
+namespace Wallet {
+
+StakeUnlockResult::~StakeUnlockResult() {}
+
+StakeUnlockResultImpl::StakeUnlockResultImpl(tools::wallet2::request_stake_unlock_result& res)
+    : result(res)
```

Comment:
The constructor argument here should just be a plain value, rather than a reference (also see comment below about making the member variable a value instead of a reference), because the reference is likely to expire before this StakeUnlockResultImpl instance does (in which case the stored reference would become invalid).

When taking it as a value you can then also use `: result(std::move(res))` to more efficiently move the `res` values into the local member variable (which allows optimizations like transferring ownership of a string's memory from one string to another, rather than needing to copy it).